common/resource/resourceImpl.go

```diff
@@ -192,7 +193,16 @@ func New(
 
 	persistenceBean, err := persistenceClient.NewBeanFromFactory(persistenceClient.NewFactory(
 		&params.PersistenceConfig,
-		persistenceMaxQPS,
+		func(...dynamicconfig.FilterOption) int {
+			if persistenceGlobalMaxQPS() > 0 {
```

Comment:
Would be awesome if we are able to exercise this code path in integration test. Checkout host/dynamicconfig.go